### DISCUSSION TITLE
How to migrate static routes to Consul dynamic routing having SignalR `wss` routes?

### DISCUSSION THREAD

**Discussion by elenacalugaru** [ThreeMammals/Ocelot#2017] [Q&A]
Hello,
So currently we have an API Gateway with Ocelot with manual routes and I have routes for SignalR like
```json
{
            "UpstreamHttpMethod": [ "GET", "POST", "PUT", "DELETE", "OPTIONS" ],
            "UpstreamPathTemplate": "/api/A/notify",
            "DownstreamPathTemplate": "/api/A/notify",
            "DownstreamScheme": "wss",
            "DownstreamHostAndPorts": [
                {
                    "Host": "{Notification}"
                }
            ],
            "SwaggerKey": "notification",
            "AddHeadersToRequest": {
                "Authorization": "Bearer {token}"
            }
        },
        {
            "UpstreamHttpMethod": [ "GET", "POST", "PUT", "DELETE", "OPTIONS" ],
            "UpstreamPathTemplate": "/api/A/notify/{catchall}",
            "DownstreamPathTemplate": "/api/A/notify/{catchall}",
            "DownstreamScheme": "wss",
            "DownstreamHostAndPorts": [
                {
                    "Host": "{Notification}"
                }
            ],
            "SwaggerKey": "notification",
            "AddHeadersToRequest": {
                "Authorization": "Bearer {token}"
            }
        }
```
and everything works well.

Now we want to switch from manual routes to dynamic routing using Consul. But in this case how can we still have those routes for SignalR to be able to connect to?

Thanks a lot,
Elena

**Comment by raman-m** [comment]
Elena, I'm just curious...
You have only 2 static routes.
Why do you want to use dynamic routes in Consul with mostly high number of them?
Maybe better to keep current configuration with 2 routes static, to be as-is?
Supporting hybrid setup is not easy.
You need to think twice before migration and preparing hybrid infrastructure.

**Reply by elenacalugaru** [reply]
Hey, I don't have only 2 static routes, i have more , this was just an example, so the question is how can I still use those routes using dynamic routing with Consul and ocelot.
We want to migrate to Consul dynamic route with all the routes so we can get rid of all the ocelot.ServiceName.json files . 
But right now we faced this issue because when using SignalR with Ocelot with manual routing it was needed to add those 2 routes and I don't know right know how to tell Consul to discover also those routes.

**Reply by raman-m** [reply]
@elenacalugaru 
You could ask our Consul and Service Discovery expert, @ggnaegi to get some dev paths.
He will provide you recommendations on the best options to migrate.
Sorry, I am too busy with development and management...
Good luck!
